pyw3d/objects.py

```diff
@@ -252,8 +252,94 @@ def fromXML(content_root):
         raise BadW3DXML("No known child node found in Content node")
 
+class W3DShape(W3DContent):
+    """ Create a shape object in virtual space
+
+    :param str shape_type: type of shape to be created
+    :param int radius: radius of the shape to be added
+    :param tuple color: Three floats representing RGB color of object 
+    :param int depth: height of shape, if shape is cone/cylinder
+    """
+    ui_order = ["shape_type","radius", "rotation", "depth"]
+    argument_validators = {
+        "shape_type": OptionValidator("Sphere", "Cube", "Cone", "Cylinder", "Monkey"),
+        "radius": IsNumeric(),
+        "rotation":ListValidator(IsInteger(min_value=0, max_value=255), required_length=3),
+        "depth": IsNumeric()
+    }
+    default_arguments = {
+        "radius": 1,
+        "rotation": (0,0,0),
+        "depth":2
+    }
+    ui_order = ["shape_type","radius", "rotation", "depth"]
+
+    def toXML(self, object_root):
+
+        content_root = ET.SubElement(object_root, "Content")
+        attrib = {
+            "shape_type": self["shape_type"],
+            "radius": self["radius"],
+            "rotation": self["rotation"],
+            "depth": self["depth"]
+        }
+        content_root = ET.SubElement(object_root, "Content")
+        shape_root = ET.SubElement(
+            content_root, "Shape")
+        shape_root.attrib["rotation"]= "{},{},{}".format(*self["rotation"])
+        shape_root.attrib["radius"] = str(self["radius"])
+        shape_root.attrib["shape_type"] = str(self["shape_type"])
+        shape_root.attrib["depth"] = str(self["depth"])
+        
+
+        #if self["shape_type"] == "Sphere":
+         #   ET.SubElement(shape_root, "Sphere")
+        #if self["shape_type"] == "Cube":
+         #   ET.SubElement(shape_root, "Cube")  
+    @classmethod
+    def fromXML(shape_class, content_root):
+        new_shape = shape_class()
+        shape_root = content_root.find("Shape")
+        if shape_root is None:
+            raise BadW3DXML("ShapeRoot element has no Shape subelement")
+        if shape_root is not None:
+            if "radius" in shape_root.attrib:
+                new_shape["radius"] = float(shape_root.attrib["radius"])                   
+            if "shape_type" in shape_root.attrib:
+                new_shape["shape_type"] = str(shape_root.attrib["shape_type"])
+        #add a child for cylinder to account for depth
+            if "depth" in shape_root.attrib:
+                new_shape["depth"] = float(shape_root.attrib["depth"])
+            if "rotation" in shape_root.attrib:
+                new_shape["rotation"] = text2tuple(shape_root.attrib["rotation"], evaluator=int)
+        
+            return new_shape       
+
+        raise InvalidArgument(
+            "Content node must contain Light node to create W3DLight object")
+
+
+    def blend(self):
+        print(self)
+       # self["shape_type"] = "Cube"
+        if self["shape_type"] == "Sphere":
```

Comment:
As noted above, we should drop the rotation here and just use a sensible default for each shape. Actual user-specified rotation should be specified by W3DObject's properties.